### PR TITLE
Update Fraction.java

### PR DESCRIPTION
**Pull request by JoshuaOsborneCYEN**

### CODE REVIEW

File: Fraction.java [hunk 1]
```diff
@@ -9,18 +9,19 @@
...
-    private int numerator;
-    private int denominator;
+    public int numerator;
+    public int denominator;
 
     public Fraction()
     {
-	numerator = 1;
-	denominator = 1;
+	numerator = (int) 1;
```
Comment (CTLSonarQubeScanner) [inline Fraction.java]: MINOR Remove this unnecessary cast to "int". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1905)

File: Fraction.java [hunk 2]
```diff
@@ -9,18 +9,19 @@
...
+	denominator = (int) 1;
```
Comment (CTLSonarQubeScanner) [inline Fraction.java]: MINOR Remove this unnecessary cast to "int". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1905)

File: Fraction.java [hunk 3]
```diff
@@ -9,18 +9,19 @@
...
     }
 
     public Fraction(int a, int b)
     {
 	setNum(a);
+	    if (1==1);
```
Comment (CTLSonarQubeScanner) [inline Fraction.java]: MAJOR Identical sub-expressions on both sides of operator "==" [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1764)
MINOR Remove this empty statement. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

File: Fraction.java [hunk 4]
```diff
@@ -9,18 +9,19 @@
 */
 public class Fraction
 {
-    private int numerator;
-    private int denominator;
+    public int numerator;
```
Comment (CTLSonarQubeScanner) [inline Fraction.java]: MINOR Make numerator a static final constant or non-public and provide accessors if needed. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: Fraction.java [hunk 5]
```diff
@@ -9,18 +9,19 @@
...
+    public int denominator;
```
Comment (CTLSonarQubeScanner) [inline Fraction.java]: MINOR Make denominator a static final constant or non-public and provide accessors if needed. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

**Comment by CTLSonarQubeScanner** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 8 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Fraction.java#L83](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L83): Use "Integer.toString" instead. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS2131)
1. ![MAJOR][MAJOR] [Fraction.java#L88](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L88): Use "Integer.toString" instead. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS2131)
1. ![MAJOR][MAJOR] [Fraction.java#L104](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L104): Equality tests should not be made with floating point values. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1244)
1. ![MAJOR][MAJOR] [Fraction.java#L134](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L134): Refactor the code in order to not assign to this loop counter from within the loop body. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)
1. ![MAJOR][MAJOR] [Fraction.java#L148](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L148): Refactor the code in order to not assign to this loop counter from within the loop body. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)
1. ![MAJOR][MAJOR] [Fraction.java#L188](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L188): Put single-quotes around '.' to use the faster "indexOf(char)" method. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS3027)
1. ![MINOR][MINOR] [Fraction.java](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java): Move this file to a named package. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1220)
1. ![MINOR][MINOR] [Fraction.java#L92](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L92): This class overrides "equals()" and should therefore also override "hashCode()". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1206)
1. ![MINOR][MINOR] [Fraction.java#L198](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L198): Use a StringBuilder instead. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1643)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by JoshuaOsborneCYEN** [comment]
Leave a comment

File: Fraction.java (hunk 1 above)
Comment (MyCodeScanner) [inline Fraction.java]: MINOR Remove this unnecessary cast to "int". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1905)

File: Fraction.java (hunk 2 above)
Comment (MyCodeScanner) [inline Fraction.java]: MINOR Remove this unnecessary cast to "int". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1905)

File: Fraction.java (hunk 3 above)
Comment (MyCodeScanner) [inline Fraction.java]: MAJOR Identical sub-expressions on both sides of operator "==" [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1764)
MINOR Remove this empty statement. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

File: Fraction.java (hunk 4 above)
Comment (MyCodeScanner) [inline Fraction.java]: MINOR Make numerator a static final constant or non-public and provide accessors if needed. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: Fraction.java (hunk 5 above)
Comment (MyCodeScanner) [inline Fraction.java]: MINOR Make denominator a static final constant or non-public and provide accessors if needed. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

**Comment by MyCodeScanner** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 8 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Fraction.java#L83](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L83): Use "Integer.toString" instead. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS2131)
1. ![MAJOR][MAJOR] [Fraction.java#L88](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L88): Use "Integer.toString" instead. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS2131)
1. ![MAJOR][MAJOR] [Fraction.java#L104](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L104): Equality tests should not be made with floating point values. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1244)
1. ![MAJOR][MAJOR] [Fraction.java#L134](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L134): Refactor the code in order to not assign to this loop counter from within the loop body. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)
1. ![MAJOR][MAJOR] [Fraction.java#L148](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L148): Refactor the code in order to not assign to this loop counter from within the loop body. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)
1. ![MAJOR][MAJOR] [Fraction.java#L188](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L188): Put single-quotes around '.' to use the faster "indexOf(char)" method. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS3027)
1. ![MINOR][MINOR] [Fraction.java](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java): Move this file to a named package. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1220)
1. ![MINOR][MINOR] [Fraction.java#L92](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L92): This class overrides "equals()" and should therefore also override "hashCode()". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1206)
1. ![MINOR][MINOR] [Fraction.java#L198](https://github.com/JoshuaOsborneCYEN/Fraction_Calculator/blob/35308eddf549d0b8f77fec69cd430c136944be81/Fraction.java#L198): Use a StringBuilder instead. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1643)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**